pkg/controller/integration/kits.go

```diff
@@ -257,10 +259,32 @@ func matchesTrait(it map[string]interface{}, kt map[string]interface{}) bool {
 }
 
 func hasMatchingSources(it *v1.Integration, kit *v1.IntegrationKit) bool {
```

Comment:
We should rename this to clearly state is used in native situations only, ie, `hasMatchingSourceForNative` and ideally add a short comment declaring the intent of this method.